core/tests/test_images.py

```diff
@@ -195,10 +195,26 @@ def test_image_save(tmp_path):
     assert_action_performed_with(image, "save", path=save_path)
 
 
-def test_as_format_toga(app):
-    """as_format can successfully return self"""
-    toga_image = toga.Image(ABSOLUTE_FILE_PATH)
-    assert toga_image is toga_image.as_format(toga.Image)
+class ImageSubclass(toga.Image):
+    pass
+
+
+@pytest.mark.parametrize(
+    "class_1, class_2",
+    [
+        (toga.Image, toga.Image),
+        (toga.Image, ImageSubclass),
+        (ImageSubclass, toga.Image),
+        (ImageSubclass, ImageSubclass),
+    ],
+)
+def test_as_format_toga(app, class_1, class_2):
```

Comment:
As a code style thing - this is one of those rare occasions where a capitalised variable name is OK. Using `Class1` and `Class2` makes it a little more syntactically obvious that you're dealing with a type (effectively a *very* weak form of [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation)).

Reply:
Thanks, good point!